File: src/service/pet/petRequest.ts

export const petRequest = () => {
  const convertFromHttpBody = (body) => {
    const { name, specie, gender, size, age, coatLength, behavior } = body
    return { name, specie, gender, size, age, coatLength, behavior }
  }

  return{ convertFromHttpBody }
}

File: src/main/java/br/com/adote/um/pet/dto/PetRequest.java
package br.com.adote.um.pet.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class PetRequest {
    private String name;
    private String specie;
    private String gender;
    private String size;
    private String age;
    private String coatLength;
    private String behavior;
}
